Adding an option to limit the number of sphinx processes.

soburi opened the pull request:
If we build with the default settings, sphinx will be run with `-j auto` specified, which will generate the same number of processes as the number of processors.
Depending on the environment, this can be a very high load.

Introduces the variable `SPHINX_PROCESSES` to specify the number of parallels so that it can be specified at the command line execution.

Such like
```
make SPHINX_PROCESSES=1
```

Fix #70712

gmarull reviewed (changes requested):
this needs to be available from CMake

kartben commented on doc/Makefile:
```diff
@@ -4,7 +4,8 @@
 BUILDDIR ?= _build
 DOC_TAG ?= development
-SPHINXOPTS ?= -j auto -W --keep-going -T
+SPHINX_PROCESSES ?= auto
```
nit: `SPHINX_JOBS` (`JOBS` would probably be fine too IMO)

soburi replied:
Thank you for the suggestion, and I applied it.

kartben reviewed (commented):
what Gerard said, + a suggestion to rename the option
Thanks for taking the time for sending this PR, I think many people will appreciate it :)

jukkar commented:
It would be great if this gets merged, currently it is almost impossible to build docs locally with my laptop.

soburi committed:
doc: Avoid redefining default values in Makefile

Pass values by setting `-D...` command arguments to the CMake only for
explicitly given options.

Changes to allow the value of DTS_ROOTS to be specified from the make
command because DTS_ROOTS is declared as a cache entry.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

soburi commented:
@gmarull 

> this needs to be available from CMake

I moved the settings to the CMake side.
Additionally, I changed to use the default value defined in CMakeLists.txt instead of redefining the defaults in the Makefile.
(Not set -D... option if no needed)

soburi commented:
@kartben 
Please check again.

henrikbrixandersen commented:
I normally just use `make SPHINXOPTS_EXTRA=-j1 html-fast` or the like.

soburi committed:
doc: Makefile: Add description how to limiting processes

By setting SPHINXOPTS_EXTRA to -j1, you can limit the number of
processes and make it easier to build on less powerful machines.
It seems that many people are not aware of this, so I have provided
a simple explanation.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

soburi commented:
@henrikbrixandersen 

Thank you for commenting.

roger that.
Looking at the conversation here, it seems that there are many people who don't know this, so I've written it as a comment.

Since I'm deleting duplicates in Makefile and CMakeLists.txt in this PR, I would like to take this opportunity to cleanup it.

MaureenHelm commented:
@kartben please revisit
